lib/client-upgrade.js

```diff
@@ -57,9 +57,9 @@ class UpgradeHandler extends AsyncResource {
 
     if (callback) {
       this.callback = null
-      process.nextTick((self, callback, err, opaque) => {
-        self.runInAsyncScope(callback, null, err, { opaque })
-      }, this, callback, err, opaque)
+      util.queueMicrotask(() => {
```

Comment:
Did you try if `.bind` instead of the array function makes a performance difference?